controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -775,11 +787,34 @@ public CompletableFuture<CreateStreamStatus.Status> createStream(String scope, S
                                                                      long createTimestamp, long requestId) {
         log.debug(requestId, "createStream with resource called.");
         OperationContext context = streamMetadataStore.createStreamContext(scope, stream, requestId);
-
-            return execute(
+        if (config.getRetentionPolicy() == null && this.globalRetentionPolicy.get()) {
+            config = createRetentionPolicy(config);
+        }
+        final StreamConfiguration  streamConfig = config;
```

Comment:
Why to create new StreamConfiguration object? why can't we use **config** itself?

Reply:
No, after mutating the `config` object it can't be passed to next method. Hence creating a new object is necessary